graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/dml/DmlFetcher.java

```diff
@@ -43,36 +43,18 @@ protected DmlFetcher(
     for (Map.Entry<String, Map<String, Object>> clauseEntry : columnList.entrySet()) {
       Column column = getColumn(table, clauseEntry.getKey());
 
-      for (Map.Entry<String, Object> condition : clauseEntry.getValue().entrySet()) {
-        if (condition.getKey().equals("in")) {
-          clause.add(
-              Condition.column(column.name()).in(buildListLiterals(column, condition.getValue())));
-          continue;
-        }
-
-        Term dbValue = toCqlTerm(column, condition.getValue());
-        switch (condition.getKey()) {
-          case "eq":
-            clause.add(Condition.column(column.name()).isEqualTo(dbValue));
-            break;
-          case "notEq":
-            clause.add(Condition.column(column.name()).isNotEqualTo(dbValue));
-            break;
-          case "gt":
-            clause.add(Condition.column(column.name()).isGreaterThan(dbValue));
-            break;
-          case "gte":
-            clause.add(Condition.column(column.name()).isGreaterThanOrEqualTo(dbValue));
-            break;
-          case "lt":
-            clause.add(Condition.column(column.name()).isLessThan(dbValue));
-            break;
-          case "lte":
-            clause.add(Condition.column(column.name()).isLessThanOrEqualTo(dbValue));
-            break;
-          default:
-            break;
+      for (Map.Entry<String, Object> conditionMap : clauseEntry.getValue().entrySet()) {
+        FilterOperator operator = FilterOperator.fromFieldName(conditionMap.getKey());
+        Condition condition;
+        if (operator != FilterOperator.IN) {
+          condition =
+              operator.buildCondition(column.name(), toCqlTerm(column, conditionMap.getValue()));
+        } else {
+          condition =
+              operator.buildCondition(
+                  column.name(), buildListLiterals(column, conditionMap.getValue()));
```

Comment:
Have you considered pushing more logic into the enum? Currently there are two `buildCondition` variant, but each constant implements exactly one of them, it's up to the caller to build the right-hand term and call the right method.

We could have single enum method:
```java
Condition buildCondition(String columnName, Object value)
```
And then each operator parses the value into terms according to what it expects (essentially moving `toCqlTerm` and `buildListLiterals` as internal utility methods in the enum).

This would allow us to write this block without any special cases:
```java
      for (Map.Entry<String, Object> conditionMap : clauseEntry.getValue().entrySet()) {
        FilterOperator operator = FilterOperator.fromFieldName(conditionMap.getKey());
        clause.add(operator.buildCondition(column.name(), conditionMap.getValue()));
      }
```

Reply:
Yeah, initially I was going in that direction but passing `nameMapping` reference put me off.

I'll give it another try to see how that goes.

Reply:
I've moved the logic to the enum, I'm happy with the result.